Rename validate - BaseModel override in stravalib model.py

In the stravalib model module, there is a method called validate that seems to override some of the expected functionality of `Basemodel.validate`. You can see the [discussion here](https://github.com/stravalib/stravalib/pull/387#discussion_r1286798070). It might make the most sense to rename this method to avoid the conflict with BaseModel. For the time being i've told mypy to ignore checking this method until the fix is in place. 


```python
class SubscriptionCallback(
    BackwardCompatibilityMixin, DeprecatedSerializableMixin
):
    """
    Represents a Webhook Event Subscription Callback.
    """

    hub_mode: Optional[str] = None
    hub_verify_token: Optional[str] = None
    hub_challenge: Optional[str] = None

    # TODO: error: Required dynamic aliases disallowed  [pydantic-alias]
    class Config:
        alias_generator = lambda field_name: field_name.replace("hub_", "hub.")

    # TODO: Signature of "validate" incompatible with supertype "BaseModel"  
    # [override] - we should consider renaming this method??
    @no_type_check
    def validate(
        self, verify_token: str = Subscription.VERIFY_TOKEN_DEFAULT
    ) -> None:
```

--- a/changelog.md
+++ b/changelog.md
@@ -6,7 +6,7 @@
 - Update client's stream method to warn when using unofficial parameters (@enadeau, #385)
 - Type annotation to client file (@enadeau, #384)
 - Fix docstring in SleepingRateLimitRule (@enadeau)
-- fix: rename `SubscriptionCallback.validate` -> `SubscriptionCallback.validate_token` to avoid conflict with `pydantic.BaseModel` (@lwasser, #396)
+- fix: rename `SubscriptionCallback.validate` -> `SubscriptionCallback.validate_token` to avoid conflict with `pydantic.BaseModel` (@lwasser, #394)
 
 ## v1.3.3
 

--- a/changelog.md
+++ b/changelog.md
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- fix: rename `SubscriptionCallback.validate` -> `SubscriptionCallback.validate_token` to avoid conflict with `pydantic.BaseModel` (@lwasser, #396)
+
 ## v1.3.3
 
 ### Fixed

--- a/stravalib/client.py
+++ b/stravalib/client.py
@@ -1720,7 +1720,7 @@ class Client(object):
 
         """
         callback = model.SubscriptionCallback.deserialize(raw)
-        callback.validate(verify_token)
+        callback.validate_token(verify_token)
         response_raw = {"hub.challenge": callback.hub_challenge}
         return response_raw
 

--- a/stravalib/model.py
+++ b/stravalib/model.py
@@ -912,7 +912,7 @@ class SubscriptionCallback(
     class Config:
         alias_generator = lambda field_name: field_name.replace("hub_", "hub.")
 
-    def validate(self, verify_token=Subscription.VERIFY_TOKEN_DEFAULT):
+    def validate_token(self, verify_token=Subscription.VERIFY_TOKEN_DEFAULT):
         assert self.hub_verify_token == verify_token
 
 

